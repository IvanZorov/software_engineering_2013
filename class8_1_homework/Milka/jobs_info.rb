require 'csv'
i=0
sum=0
age=0
array = Array.new()
CSV.foreach(ARGV[0]) do |row|
	if ARGV[1] == row[2]
		array << row		
		i+=1
		sum+=row[3].to_f
		age+=row[1].to_i
	end
end
sum = sum/i
age = age/i

array.sort!{|a,b| a[3].to_f <=> b[3].to_f}
array << [age,sum.round(2)]
CSV.open("jobs_info_result.csv","w") do |csv|
	array.each do |element|
	csv << element
	end

end
