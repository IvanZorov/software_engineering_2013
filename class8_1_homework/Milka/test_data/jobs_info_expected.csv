Chastity Budd,25,programer,1585.00
Robbi Liang,27,programer,1674.35
Kellie Paulin,23,programer,2400.00
Samella Tufts,27,programer,2410.00
Charita Woomer,40,programer,3630.53
Glenda Endo,23,programer,5499.24
Boris Delafuente,23,programer,6400.50
Magaret Caesar,34,programer,60590.20
27,10523.73
